### PR TITLE
Main typing

### PR DESCRIPTION
**Pull request by vasilyatka**
Hey. I've tested it in TS project and it looks ok.

### CODE REVIEW

**Comment by localjo** [comment]
Hi @vasilyatka thank you for the contribution. Can you explain why the options type should be exported separately from the props, and why `React.ReactNode` should be used instead of `React.Element`?

**Comment by vasyan** [comment]
I see, maybe TaterOptions not necessary. 
What about ReactNode and React.Element is that ReactNode includes itself all valid ReactNode, lol. Check this, please https://github.com/DefinitelyTyped/DefinitelyTyped/blob/5344bfc80508c53a23dae37b860fb0c905ff7b24/types/react/index.d.ts#L150
While React.Element is an object with fields 'type', 'props', and 'key'. That isn't good for the case when you pass to children just a string.

**Comment by localjo** [comment]
Interesting. In this case, since the library is designed to wrap an element that has a large enough space for putting annotations on it, I think `React.Element` might be better. This library wouldn't work if the user was trying to wrap other elements like `ReactFragment | ReactPortal | string | number | boolean | null | undefined`.

**Comment by vasyan** [comment]
Any news?

**Comment by localjo** [comment]
What are the problems with the current implementation, and how does this PR solve them?

**Comment by vasyan** [comment]
The problem is with options prop. The current implementation has a mistyping with it. Check it, please
image

**Comment by localjo** [comment]
Try passing the props directly as `name` and `size`. There's no `options` prop.

**Comment by vasyan** [comment]
Actually it is. Look please at this line
https://github.com/localjo/react-tater/blob/90cdbe811133c0e710a3b298183d238e40c3283a/src/index.js#L31

Now API is not correct

File: src/index.d.ts [hunk 1]
```diff
@@ -1,13 +1,18 @@
...
+  space?: number;
+}
 
-  export default Tater
+interface TaterProps {
+  options?: TaterOptions;
+  children: ReactElement;
 }
+
+declare const Tater: (props: TaterProps) => JSX.Element;
```
Comment (localjo) [inline src/index.d.ts:13]: Why is the return type `JSX.Element` instead of `ReactElement`?

Comment (localjo) [reply src/index.d.ts:13]: Ah, I see https://stackoverflow.com/questions/58123398/when-to-use-jsx-element-vs-reactnode-vs-reactelement

**Pull request merged**